microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/services/MicroProfileDocumentHighlight.java

```diff
@@ -0,0 +1,116 @@
+		try {
+			Node node = document.findNodeAt(position);
+			switch (node.getNodeType()) {
+				case PROPERTY_KEY:
+					return propertyKeyHighlight(document, (PropertyKey) node);
+				case PROPERTY_VALUE_EXPRESSION:
+					return propertyValueExpressionHighlight(document, (PropertyValueExpression) node);
+				default:
+					return Collections.emptyList();
+			}
+		} catch (BadLocationException e) {
+			return Collections.emptyList();
+		}
+	}
+
+	/**
+	 * Highlight the key portion of the assignment statement for the property that
+	 * the property expression references, or nothing
+	 *
+	 * @param document The properties model that is being worked with
+	 * @param node     The property value expression that the cursor is on
+	 * @return A singleton list that contains a highlight for the key in the
+	 *         assignment statement for the reference property, or an empty list
+	 */
+	private List<? extends DocumentHighlight> propertyValueExpressionHighlight(PropertiesModel document,
+			PropertyValueExpression node) {
+		String otherProp = node.getReferencedPropertyName();
+		if (otherProp == null || otherProp.isBlank()) {
+			return Collections.emptyList();
+		}
+		for (Node child : document.getChildren()) {
+			if (child.getNodeType() == NodeType.PROPERTY) {
+				Property property = (Property) child;
+				if (otherProp.equals(property.getPropertyName())) {
+					return Collections
+							.singletonList(new DocumentHighlight(PositionUtils.createRange(property.getKey()), DocumentHighlightKind.Write));
+				}
+			}
+		}
+		return Collections.emptyList();
+	}
+
+	/**
+	 * Highlight all the property expressions that reference the property that the
+	 * given property key represents
+	 *
+	 * @param document The properties model that is being worked with
+	 * @param node     The property key that the cursor is on
+	 * @return A list containing highlights for all property expressions in the
+	 *         document that reference the expression that the property key
+	 *         represents
+	 */
+	private List<? extends DocumentHighlight> propertyKeyHighlight(PropertiesModel document, PropertyKey node) {
+		String propertyName = node.getPropertyName();
+		if (propertyName == null || propertyName.isBlank()) {
+			return Collections.emptyList();
+		}
+		List<DocumentHighlight> highlights = new ArrayList<>();
+		for (Node child : document.getChildren()) {
+			if (child.getNodeType() == NodeType.PROPERTY) {
+				Property property = (Property) child;
+				if (property.getValue() != null) {
+					for (Node valueSegment : ((Property) child).getValue().getChildren()) {
+						if (valueSegment.getNodeType() == NodeType.PROPERTY_VALUE_EXPRESSION && propertyName
+								.equals(((PropertyValueExpression) valueSegment).getReferencedPropertyName())) {
+							highlights.add(new DocumentHighlight(PositionUtils.createRange(valueSegment), DocumentHighlightKind.Read));
```

Comment:
You should highlight the referenced property which have the cursor as `DocumentHighlightKind.Write` too.